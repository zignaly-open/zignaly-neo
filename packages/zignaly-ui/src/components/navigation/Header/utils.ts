// TODO: Move this for config or theme folder.
export const headerConfig = {
  height: "52px",
};
