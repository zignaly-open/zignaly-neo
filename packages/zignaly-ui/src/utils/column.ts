import styled from "styled-components";

export const Column = styled.div<{
  gap?: number;
  justifyContent?: "center" | "end" | "start" | "space-between";
  alignItems?: "center" | "end" | "start" | "space-between";
  textAlign?: "end" | "center" | "start";
  height?: string;
  flex?: number;
  width?: string;
}>`
  display: flex;
  flex-direction: column;
  flex: ${(props) => props.flex};
  gap: ${(props) => props.gap}px;
  justify-content: ${(props) => props.justifyContent};
  align-items: ${(props) => props.alignItems};
  text-align: ${(props) => `${props.textAlign}`};
  height: ${(props) => `${props.height}`};
  width: ${(props) => `${props.width}`};
`;
