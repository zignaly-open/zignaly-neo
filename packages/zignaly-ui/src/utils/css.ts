import { css } from "styled-components";

export const NiceScrollbar = css`
  &::-webkit-scrollbar {
    width: 6px;
  }

  /* Track */
  &::-webkit-scrollbar-track {
    background: rgba(255, 255, 255, 0.05);
    border-radius: 12px;
  }

  /* Handle */
  &::-webkit-scrollbar-thumb {
    background: rgba(255, 255, 255, 0.1);
    border-radius: 12px;
  }

  /* Handle on hover */
  &::-webkit-scrollbar-thumb:hover {
    background: rgba(255, 255, 255, 0.15);
  }
`;
