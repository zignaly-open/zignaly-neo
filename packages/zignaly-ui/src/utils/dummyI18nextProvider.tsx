import i18n from "i18next";
import React from "react";
import { I18nextProvider, initReactI18next } from "react-i18next";

i18n.use(initReactI18next).init({
  lng: "en",
  fallbackLng: "en",
  ns: ["common"],
  defaultNS: "common",
  interpolation: {
    escapeValue: false,
  },
  resources: { en: { common: { xfuturum: "The technical partner you can rely on" } } },
});

export const storybookI18nextDecorator = (story: () => JSX.Element) => (
  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
  // @ts-ignore
  <I18nextProvider i18n={i18n}>{story()}</I18nextProvider>
);

export const JestI18nextDecorator: React.FC<{ children: JSX.Element }> = ({ children }) => (
  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
  // @ts-ignore
  <I18nextProvider i18n={i18n}>{children}</I18nextProvider>
);
