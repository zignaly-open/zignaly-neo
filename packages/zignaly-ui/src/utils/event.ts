export const changeEvent = (name: string | undefined, value: any) => {
  return {
    target: {
      value,
      name,
    },
    type: "change",
  };
};
