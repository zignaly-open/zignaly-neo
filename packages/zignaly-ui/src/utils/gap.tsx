import styled from "styled-components";

export const Gap = styled.div<{ gap: number }>`
  ${(props) => `
  padding: ${props.gap}px;
  `}
`;
