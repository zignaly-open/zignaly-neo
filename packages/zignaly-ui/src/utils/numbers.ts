export const sortByValue = (number1: string | number, number2: string | number) => {
  return +number1 - +number2;
};

export const trimZeros = (v: string | number) => {
  return v?.toString()?.replace(/(\.\d*?[1-9]+)0+$|\.0+$/g, "$1");
};

export const numberOfDecimals = (v: string | number) => {
  return (v?.toString()?.split(".")[1] ?? "").length;
};

export const formatCompactNumber = (value: number | string, precision = 8) =>
  new Intl.NumberFormat("en-US", {
    maximumFractionDigits: precision,
    notation: "compact",
    compactDisplay: "short",
  }).format(+value);
