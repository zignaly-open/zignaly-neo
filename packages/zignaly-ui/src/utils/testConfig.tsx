import { render } from "@testing-library/react";
import { ThemeProvider as ThemeProviderMui } from "@mui/material";
import { ThemeProvider, ThemeProviderMui as ThemeInheritorMui } from "../index";
import React from "react";
import { getZignalyThemeExport } from "../theme";
import mediaQuery from "css-mediaquery";
import { JestI18nextDecorator } from "./dummyI18nextProvider";

export const renderWithProvidersUi = (ui: JSX.Element, renderOptions = {}) => {
  const { mui, legacyStyledComponentsDoNotUse } = getZignalyThemeExport();
  return render(ui, {
    wrapper: ({ children }: { children: JSX.Element }) => (
      <JestI18nextDecorator>
        <ThemeProvider theme={legacyStyledComponentsDoNotUse}>
          <ThemeInheritorMui theme={mui}>
            <ThemeProviderMui theme={mui}>{children}</ThemeProviderMui>
          </ThemeInheritorMui>
        </ThemeProvider>
      </JestI18nextDecorator>
    ),
    ...renderOptions,
  });
};

const createMatchMedia = (width: number) => {
  return (query: string) => {
    return {
      matches: mediaQuery.match(query, { width }),
      media: "",
      addListener: () => {},
      removeListener: () => {},
      onchange: () => {},
      addEventListener: () => {},
      removeEventListener: () => {},
      dispatchEvent: () => true,
    };
  };
};

export const resizeScreenSize = (width: number) => {
  window.matchMedia = createMatchMedia(width);
};
