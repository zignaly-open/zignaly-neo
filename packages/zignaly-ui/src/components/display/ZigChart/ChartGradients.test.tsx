import React from "react";
import { ChartGradients } from ".";
import { renderWithProvidersUi } from "../../../utils/testConfig";

describe("ChartGradients component tests", () => {
  it("renders ChartGradients component", () => {
    const { container } = renderWithProvidersUi(<ChartGradients />);
    expect(container).toMatchSnapshot();
  });
});
