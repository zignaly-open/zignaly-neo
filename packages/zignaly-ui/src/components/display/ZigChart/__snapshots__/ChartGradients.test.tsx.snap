// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChartGradients component tests renders ChartGradients component 1`] = `
<div>
  <svg
    style="width: 0px; height: 0px; display: block;"
  >
    <defs>
      <lineargradient
        id="chart-gradient-mini-red"
        x1="1%"
        x2="0%"
        y1="100%"
        y2="0%"
      >
        <stop
          offset="0%"
          stop-color="rgba(18, 20, 39, 0)"
        />
        <stop
          offset="50%"
          stop-color="rgba(21, 21, 57, 0.5)"
        />
        <stop
          offset="100%"
          stop-color="rgba(86, 36, 108, 1)"
        />
      </lineargradient>
      <lineargradient
        id="chart-gradient-mini-green"
        x1="1%"
        x2="0%"
        y1="100%"
        y2="0%"
      >
        <stop
          offset="0%"
          stop-color="rgba(17, 27, 47, 0)"
        />
        <stop
          offset="50%"
          stop-color="rgba(22, 41, 67, 0.5)"
        />
        <stop
          offset="100%"
          stop-color="rgba(39, 110, 107, 1)"
        />
      </lineargradient>
      <lineargradient
        id="chart-gradient-card-red"
        x1="0"
        x2="0"
        y1="1"
        y2="0"
      >
        <stop
          offset="0%"
          stop-color="rgba(37, 15, 38, 0.38)"
          stop-opacity="0.76"
        />
        <stop
          offset="32%"
          stop-color="rgba(46, 0, 28, 0.44)"
          stop-opacity="0.76"
        />
        <stop
          offset="100%"
          stop-color="rgba(204, 57, 147, 0.69)"
          stop-opacity="0.76"
        />
      </lineargradient>
      <lineargradient
        id="chart-gradient-card-green"
        x1="0"
        x2="0"
        y1="1"
        y2="0"
      >
        <stop
          offset="0%"
          stop-color="rgba(16, 33, 33, 0.68)"
          stop-opacity="0.76"
        />
        <stop
          offset="32%"
          stop-color="rgba(0, 45, 20, 0.44)"
          stop-opacity="0.76"
        />
        <stop
          offset="53%"
          stop-color="rgba(7, 47, 30, 0.44)"
          stop-opacity="0.76"
        />
        <stop
          offset="100%"
          stop-color="rgba(56, 235, 170, 0.69)"
          stop-opacity="0.76"
        />
      </lineargradient>
      <lineargradient
        id="chart-gradient-full-red"
        x1="0"
        x2="0"
        y1="1"
        y2="0"
      >
        <stop
          offset="0%"
          stop-color="rgba(31, 18, 59, 0.52)"
        />
        <stop
          offset="100%"
          stop-color="rgba(74, 33, 81, 0.69)"
        />
      </lineargradient>
      <lineargradient
        id="chart-gradient-full-green"
        x1="0"
        x2="0"
        y1="1"
        y2="0"
      >
        <stop
          offset="0%"
          stop-color="rgba(18, 33, 59, 0.52)"
        />
        <stop
          offset="100%"
          stop-color="rgba(33, 81, 78, 0.69)"
        />
      </lineargradient>
    </defs>
  </svg>
</div>
`;
